### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by NATASHA-ct**
### I implemented the following project requirements ✔️ 

- [x] Check the usage of [Decorator pattern in Ruby](https://refactoring.guru/design-patterns/decorator/ruby/example#example-0)
- [x] Think about how you can use two decorators in order to capitalize and trim people's names.
- [x] In the following steps you will get instructions for implementing the Decorator design pattern for this case.

#### Interface

- [x]  Create a class `Nameable`.
- [x]  Implement a method called `correct_name` that will raise a `NotImplementedError`.

#### Turn your Person class to Nameable

- [x]  Make sure that your Person class inherits from Nameable
- [x]  Make sure that this class has a method `correct_name` implemented. It should simply return the name attribute.

#### Prepare base Decorator

- [x] Make sure that it inherits from Nameable.
- [x] In the constructor assign a nameable object from params to an instance variable.
- [x] Implement the `correct_name` method that returns the result of the correct_name method of the `@nameable`.

#### Prepare CapitalizeDecorator and TrimmerDecorator

- [x] For the CapitalizeDecorator:

    - Create a class that inherits from the base Decorator class.
    - Implement a method `correct_name` that capitalizes the output of `@nameable.correct_name`.

- [x] For the TrimmerDecorator:

    - Create a class that inherits from the base Decorator class.
    - Implement a method `correct_name` that makes sure that the output of `@nameable.correct_name` has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: base_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Decorator inherits from nameable
+require_relative './nameable'
+
+class Decorator < Nameable
+  # constructor
+  def initialize(_nameable)
+    super()
+    @name = name
```
Comment (fobadara) [inline base_decorator.rb]: [REQUIRED] You are expected to assign a nameable object from params

File: base_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  end
+
+  def correct_name
+    @nameable
```
Comment (fobadara) [inline base_decorator.rb]: [REQUIRED] You are expected to call the correct_name method present in @nameable here.

File: trimmerdecorator.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+require './base_decorator'
+
+class TrimmerDecorator < Decorator
+  # method to make sure output of nameable is capitalized
+  def correct_name
+    @nameable.correct_name[0...10]
```
Comment (fobadara) [inline trimmerdecorator.rb]: [REQUIRED] Kindly change the number 10 to 9. Don't forget 0 indexing in Ruby.

**Review by fobadara — CHANGES_REQUESTED** [review]
# REQUIRES CHANGES ♻️ 
Hi @NATASHA-ct, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Highlights
✅ Descriptive README
✅ Descriptive PR summary
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by cynthiainga — APPROVED** [review]
Hi @NATASHA-ct, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [ ] No linters errors ✔️ 
- [ ] Descriptive PR ✔️ 
- [ ] Good commit history ✔️ 
- [ ] Correct Gitflow ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**